cardano-explorer/src/Explorer/Web/LegacyApi.hs

```diff
@@ -0,0 +1,155 @@
+{-# LANGUAGE DataKinds     #-}
+{-# LANGUAGE InstanceSigs  #-}
+{-# LANGUAGE TypeFamilies  #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE DeriveGeneric #-}
+
+-- | Type-level specification of Explorer API (via Servant).
+
+module Explorer.Web.LegacyApi
+       ( ExplorerApiRecord(..)
+       ) where
+
+import           Universum
+
+import           Servant.API ((:>), Capture, Get, JSON, Post, QueryParam,
+                     ReqBody, Summary)
+import           Servant.API.Generic ((:-))
+
+import           Cardano.Chain.Slotting (EpochNumber)
+import           Explorer.Web.ClientTypes (CAda, CAddress,
+                     CAddressSummary, CAddressesFilter, CBlockEntry,
+                     CBlockSummary, CGenesisAddressInfo, CGenesisSummary,
+                     CHash, CTxBrief, CTxEntry, CTxId, CTxSummary, CUtxo, CBlockRange)
+import           Explorer.Web.Error (ExplorerError)
+--import           Pos.Util.Servant (DQueryParam, ModifiesApiRes (..), VerbMod)
+
+data WithDefaultApiArg subApi
+type DQueryParam s a = WithDefaultApiArg (QueryParam s a)
+
+type PageNumber = Integer
```

Comment:
That should probably be a `newtype`, and since we are likely to have a relatively finite number of pages, `Int` would be more appropriate. If we are not going to do arithmetic on it, `Word` might be even better.